src/_posts/hands-on-iii.md

```diff
@@ -0,0 +1,334 @@
+
+One way of preserving this data, without changing the way the element is used, is to store it inside the HTML node. The important piece is in the `init` method where the component creates or reuses a storage object to hold the state.
+
+```js
+AFRAME.registerComponent('tool', {
+  schema: {
+    position: { type: 'vec3' },
+    rotation: { type: 'vec3' }
+  },
+  init() {
+    this.el._tool = this.el._tool || {};
+  },
+  grab() {
+    this._rememberLocation();
+    this.el.setAttribute('position', this.data.position);
+    this.el.setAttribute('rotation', this.data.rotation);
+    this.el._tool._isGrabbed = true;
+  },
+  drop() {
+    this._restoreLocation();
+    this.el._tool._isGrabbed = false;
+  },
+  _rememberLocation() {
+    this.el._tool._originalParent = this.el.parentNode;
+    this.el._tool._originalPosition = this.el.getAttribute('position');
+    this.el._tool._originalRotation = this.el.getAttribute('rotation');
+  },
+  _restoreLocation() {
+    this.el._tool._originalParent.appendChild(this.el);
+    this.el.setAttribute('position', this.el._tool._originalPosition);
+    this.el.setAttribute('rotation', this.el._tool._originalRotation);
+  },
+  isGrabbed() {
+    return this.el._tool._isGrabbed;
+  }
+});
+```
+
+Now we can grab every element in the scene:
+
+[![Grabbing tools](/images/blog/hands-on/step3-grabbing-tools-thumb.png)](/images/blog/hands-on/step3-grabbing-tools.gif)
+
+### Fixing the collision bounding shapes
+
+In the [previous article](/blog/hands-on-ii.html), I mentioned that the physic engine was miscalculating the collision bounding box of the VIVE controllers. Making the bounding box to match the controller model is especially important when trying to grab physical objects, after disabling the debug mode of physics. If the bounding box, and the model position and dimensions don’t match, the user could not predict the distances for which grabbing something is possible resulting in a counterintuitive and frustrating interaction.
+
+Both A-Frame and the physics component share some of the responsibility of the glitch: on one side, the physics component is not considering the current rotation of the controller properly so, unless the controller is resting in its default position, the bounding box dimensions would be miscalculated. On the A-Frame side, the `vive-controls` component is [applying a correction in the position of the model after loading the model](https://github.com/aframevr/aframe/blob/master/src/components/vive-controls.js#L202). This correction is not advertised outside the component so the physics component can not realize when the model is fully placed. Since the bounding box calculation is done before applying the correction, the bounding box will be misplaced.
```

Comment:
`The problem lies in the communication between A-Frame and the pyhsics component. The pyhsics component does not consider... On the other hand, A-Frame's vive-controls component...`